src/vse_sync_pp/parsers/parser.py

```diff
@@ -3,7 +3,57 @@
 """Common parser functionality"""
 
 import json
-from decimal import Decimal
+import re
+from datetime import (datetime, timezone)
+from decimal import (Decimal, InvalidOperation)
+
+# sufficient regex to extract the whole decimal fraction part
+RE_ISO8601_DECFRAC = re.compile(
+    r'^(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2})\.(\d+)(.*)$'
+)
+
+def parse_timestamp_abs(val):
+    """Return a :class:`Decimal` from `val`, an absolute timestamp string.
+
+    Accepted absolute timestamp strings are ISO 8601 format with restrictions.
+    The string must: explicitly specify UTC timezone, supply time in seconds,
+    specify a decimal fractional part with a decimal mark of '.'.
+
+    Return None if `val` is not a string or is not an ISO 8601 format string.
+    Raise :class:`ValueError` otherwise.
+    """
+    try:
+        dtv = datetime.fromisoformat(val)
+    except TypeError:
+        return None
+    except ValueError:
+        # before Python 3.11 `fromisoformat` fails if UTC denoted by 'Z' and/or
+        # the decimal fraction has more than 6 digits
+        match = RE_ISO8601_DECFRAC.match(val)
+        if match is None:
+            return None
+        # parse without decimal fraction, with 'Z' substituted
+        dtv = datetime.fromisoformat(
+            match.group(1) + '+00:00' if match.group(3) == 'Z' else match.group(3)
+        )
+    else:
+        match = RE_ISO8601_DECFRAC.match(val)
+        if match is None:
+            raise ValueError(val)
+    if dtv.tzinfo != timezone.utc:
+        raise ValueError(val)
+    return Decimal(f'{int(dtv.timestamp())}.{match.group(2)}')
```

Comment:
A comment about the seconds getting truncated here would be helpful.

Reply:
Happy to oblige